main.tf

```diff
@@ -18,7 +18,8 @@ resource "aws_iam_user" "default" {
 }
 
 resource "aws_iam_access_key" "default" {
```

Comment:
Maybe add in a `moved` block?

Why?

This changes the state file from `aws_iam_access_key.default` to `aws_iam_access_key.default[0]` in existing states. Resulting in recreating of the access keys.

I don't think this is wanted behaviour 😏 

This can be prevented with the `moved` thingy

Reply:
```hcl

moved {
  from = aws_iam_access_key.default
  to - aws_iam_access_key.default[0]
}
```

Something like this

Reply:
agreed resolved